app/src/utils/session.js

```diff
@@ -1,10 +1,19 @@
 import { ref } from 'vue'
+import { FileGraph } from '@linux-odyssey/file-graph'
 import api from './api'
 
 class SessionManager {
   constructor(questId) {
     this.questId = questId
     this.session = ref(null)
+    this.graph = ref(
+      new FileGraph({
+        path: '/',
+        type: 'folder',
+        discovered: false,
+      })
+    )
+    this.pwd = ref('')
   }
```

Comment:
I've seen ref() many times, how can use it in various way?

Reply:
ref creates a proxy object to a value, you can consider it as pointer. It can store anything you want, and you access its value via ref.value. Everytime a ref's value updates, all Vue components depends on it gets re-render.